common/defs/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+project(cgadefs)
+
+find_package(CUDA 9.0 QUIET REQUIRED)
+
+add_library(${PROJECT_NAME} INTERFACE)
+target_link_libraries(${PROJECT_NAME} INTERFACE ${CUDA_LIBRARIES})
+
+if (cga_profiling)
```

Comment:
I believe all the CUDA libs, profiling macros and debugging macros are taken care of in the `cgaconfig` project, so they don't need to be added again to new ones. Is that right @ahehn-nv ?

Reply:
Probably, I largely copied it from `cgaconfig`. Is it the necessary to include `cgaconfig` somehow? I'm still having problems what gets included and how in our structure.

Reply:
Just do a `target_link_libraries(${PROJECT_NAME} INTERFACE cgaconfig)`.

In general, you should only need
```
add_library(${PROJECT_NAME} INTERFACE)
target_link_libraries(${PROJECT_NAME} INTERFACE cgaconfig)target_include_directories(${PROJECT_NAME}
    INTERFACE 
        $<INSTALL_INTERFACE:include>    
        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
)

install(TARGETS ${PROJECT_NAME}
    EXPORT ${PROJECT_NAME}
    INCLUDES DESTINATION include
)
install(DIRECTORY include/ DESTINATION include)
install(EXPORT ${PROJECT_NAME} DESTINATION cmake)
# Adding formatting
cga_enable_auto_formatting("${CMAKE_CURRENT_SOURCE_DIR}")
```
here.

Reply:
We decided not to introduce `defs` library. Marking this as resolved